pkg/nmi/server/server.go

```diff
@@ -380,3 +409,46 @@ func handleTermination() {
 	log.Infof("Exiting with %v", exitCode)
 	os.Exit(exitCode)
 }
+
+func listPodIDsInCreatedStateWithRetries(ctx context.Context, kubeClient k8s.Client, logger *log.Entry, podns, podname, rqClientID string, maxAttempts int) (*[]aadpodid.AzureIdentity, error) {
+	// created and assigned states are added to statesToCheck. This is to ensure if the call is made after the assigned identity is in assigned state
+	// for the pod, we shouldn't fail on the hard check for created state. Both states are accepted values for passing first check.
+	return listPodIDsWithRetry(ctx, kubeClient, logger, podns, podname, rqClientID, []string{aadpodid.AssignedIDCreated, aadpodid.AssignedIDAssigned}, maxAttempts)
```

Comment:
Why are both state assigned ids added when the function name indicates we are going to gather Created ones.